src/libopensc/card-cac.c

```diff
@@ -1852,6 +1852,12 @@ static int cac_init(sc_card_t *card)
 	LOG_FUNC_RETURN(card->ctx, SC_SUCCESS);
 }
 
+static int cac_logout(sc_card_t *card)
+{
+	int index;
+	return cac_find_first_pki_applet(card, &index);
+}
+
```

Comment:
This is the same for CAC and CAC1 so I would propose to move it to card-cac-common.{c,h} to avoid code duplication (even though its just couple of lines)

Reply:
cac_find_first_pki_applet is implemented independent from each other (not in card-cac-common.{c,h}). If you confirm, I would move this to the common part as well

Reply:
Oh, my bad. It was long time since I touched the CAC driver and forgot there are these differences in implementation.

As far as I know, selecting any applet broke the login state in the CAC so maybe just calling select applet might do, but I would have to check or try. Let me get back to that next week.